samples/svg/svgtest.cpp

```diff
@@ -230,204 +226,213 @@ void MyFrame::FileSavePicture(wxCommandEvent& WXUNUSED(event))
 
 // Define a constructor for my page
 MyPage::MyPage(wxNotebook *parent, int index)
-    : wxScrolledWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxVSCROLL|wxHSCROLL)
+    : wxScrolledWindow(parent)
 {
     SetBackgroundColour(*wxWHITE);
     SetScrollbars(20, 20, 50, 50);
     m_index = index;
 }
 
-bool MyPage::OnSave(wxString filename)
+bool MyPage::OnSave(const wxString& filename)
 {
-    wxSVGFileDC svgDC (filename, 600, 650);
-    OnDraw (svgDC);
+    wxSize svgSize;
+    wxSVGFileDC tempSvgDC(svgSize);
+    OnDraw(tempSvgDC);
+
+    svgSize = wxSize(tempSvgDC.MaxX(), tempSvgDC.MaxY());
+    svgSize.IncBy(15); // account for wxPen width exceeding bounds
+
+    wxSVGFileDC svgDC(svgSize, filename, pageNames[m_index]);
+    svgDC.SetBitmapHandler(new wxSVGBitmapEmbedHandler());
+    OnDraw(svgDC);
+
     return svgDC.IsOk();
 }
 
 // Define the repainting behaviour
 void MyPage::OnDraw(wxDC& dc)
 {
-     // vars to use ...
+    // vars to use ...
     wxPen wP;
     wxBrush wB;
     wxPoint points[6];
     wxColour wC;
 
-    dc.SetFont(*wxSWISS_FONT);
+    dc.SetFont(wxFontInfo(9).FaceName("Arial").Family(wxFONTFAMILY_SWISS));
     dc.SetPen(*wxGREEN_PEN);
 
     switch (m_index)
     {
         case Page_Lines:
             // draw lines to make a cross
-            dc.DrawLine(0, 0, 200, 200);
-            dc.DrawLine(200, 0, 0, 200);
+            dc.DrawLine(dc.FromDIP(0), dc.FromDIP(0), dc.FromDIP(200), dc.FromDIP(200));
```

Comment:
Similarly to the remark above, maybe we should move this code into a function to make sure it doesn't use `MyPage` scaling factor?

Reply:
Done. Do you agree with this approach?

Reply:
Yes, nice, thank you!